src/routes/route.py

```diff
@@ -1,67 +1,95 @@
-        """
-        Introduces different departure times for the persons taking the given route.
 
-        :param chunks: In how many different chunks should the persons on the route depart.
-        :param interval: How long should the interval between each departure be. Given in seconds.
-        :return: A dictionary of departure times where keys are seconds after midnight and values are the number of people departing on that time.
-        """
-        departure_times: dict[int, int] = {}
-        people_per_interval = int(self.num_people_on_route / chunks)
-        remaining_people = self.num_people_on_route % chunks
+def _departure_times(total_population: int, start: int, end: int) -> NDArray[np.int_]:
+    """
+    Introduces different departure times for the persons taking the given route, spread on a normal distribution
+    between the given start and end times.
+    :param total_population: The total population.
+    :param start: start of normal distribution. Given in seconds.
+    :param end: end of normal distribution. Given in seconds.
+    """
+    mean = (start + end) / 2  # mean
+    std_dev = (end - start) / 6  # Approx. 99.7% of values within range
+    rng = np.random.default_rng()
+    departures = rng.normal(loc=mean, scale=std_dev, size=total_population).astype(
+        np.int_
+    )
+    return departures
 
-        seconds_from_midnight = 0
-        if chunks < self.num_people_on_route:
-            for i in range(chunks):
-                count = people_per_interval + (
-                    1 if i < remaining_people else 0
-                )  # Distribute remainder evenly
-                departure_times[seconds_from_midnight] = count
-                seconds_from_midnight += interval
-        else:  # There are more chunks than people, so there will be 1 person in the first chunk(s) and then 0 in the remaining. The remaining will not even be present in the departure_times dict.
-            for i in range(self.num_people_on_route):
-                departure_times[seconds_from_midnight] = 1
-                seconds_from_midnight += interval
 
-        self.departure_times = departure_times
+def _get_num_people_on_route(
+    route_path: list[str], population_data: gpd.GeoDataFrame, cars_per_person: float
+) -> int:
+    """
+    Returns the number of people on a given route.
+    :param route_path: A list of node IDs representing the route.
+    :param population_data: A GeoDataFrame containing the population data.
+    :param cars_per_person: The number of cars per person.
+    :return: The number of people on the route.
+    """
+    return int(
+        population_data[population_data[NODE_ID] == route_path[0]].iloc[0][POPULATION]
+        * cars_per_person
+    )
+
+
+def _get_total_population(
+    population_data: gpd.GeoDataFrame, cars_per_person: float
+) -> int:
+    result = population_data[POPULATION].sum()
+    if result is None:
+        logging.fatal("Population data is empty")
```

Comment:
```suggestion
    if result is None:
        logging.fatal("Population data is empty")
        raise ValueError("Population data is empty or zero.")
```

Reply:
I meant to include the line above as well. Can the `sum()` of a GDF be `None`, and should we be checking for `result == 0`?